speech-api/SpeechSynthesis-speak-events.html

```diff
@@ -5,14 +5,21 @@
 <script src="/resources/testdriver-vendor.js"></script>
 <body>
 <script>
-async_test(t => {
+function runStartEndTest(t, utterance) {
   test_driver.bless('speechSynthesis.speak', t.step_func(() => {
-    const utter = new SpeechSynthesisUtterance('test');
-    utter.onerror = t.unreached_func('error event');
-    speechSynthesis.speak(utter);
-    utter.onstart = t.step_func(() => {
-      utter.onend = t.step_func_done();
+    utterance.onerror = t.unreached_func('error event');
+    speechSynthesis.speak(utterance);
+    utterance.onstart = t.step_func(() => {
+      utterance.onend = t.step_func_done();
     });
   }));
+}
+async_test(t => {
```

Comment:
These tests will run in parallel, and given the global nature of speech synthesis that's probably not a good idea, they might interfere. Testing that somewhere is a good idea, but it's not what these tests are primarily about.

If you use `promise_test` the tests will run one after another automatically.

Using [`EventWatcher`](https://web-platform-tests.org/writing-tests/testharness-api.html?highlight=eventwatcher#eventwatcher) will probably work well here.

So something like:
```js
promise_test(async (t) => {
  const utterance = new SpeechSynthesisUtterance();
  const eventWatcher = new EventWatcher(t, utterance, ['start', 'end', 'error']);
  await test_driver.bless('speechSynthesis.speak', () => speechSynthesis.speak(utterance));
  await eventWatcher.wait_for(['start', 'end']);
}, 'speechSynthesis.speak() fires start and end events with empty utterance');
```

This is then simple enough that I wouldn't bother abstracting out a helper.

Reply:
Thanks for the tips, Philip! I addressed your comments and the tests run one after the other now. However, I left the helper function in because I think it makes less duplicated code. If you think that there is a reason, that outweighs avoiding duplicate code, for removing the helper, let me know and I will update the PR.